### PR TITLE
Use Boxes in SyntaxToken.Value

### LINKED ISSUE #51524: Unnecessary boxing in SyntaxToken.Value
These return values:

https://github.com/dotnet/roslyn/blob/b1ef0b3d0674737f9e60a701fcace2b75883e1f6/src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs#L287-L290

Should be using these boxes:

https://github.com/dotnet/roslyn/blob/b1ef0b3d0674737f9e60a701fcace2b75883e1f6/src/Compilers/Core/Portable/Collections/Boxes.cs#L11-L12

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #51524

### CODE REVIEW

**Comment by Youssef1313** [comment]
Integration tests failure is unrelated.

**Review by cston — APPROVED** [review]

**Review by jcouv — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs [hunk 1]
```diff
@@ -285,9 +286,9 @@ public virtual object Value
                 switch (this.Kind)
                 {
                     case SyntaxKind.TrueKeyword:
-                        return true;
+                        return Boxes.BoxedTrue;
                     case SyntaxKind.FalseKeyword:
-                        return false;
+                        return Boxes.BoxedFalse;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs:291]: >Boxes.BoxedFalse [](start = 31, length = 16)

Is similar change applicable to VB?

Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs:291]: @AlekseyTs VB is already using Boxes.

Comment (sharwell) [reply src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs:291]: VB already used `Boxes`:

https://github.com/dotnet/roslyn/blob/478adbc6624774e081b122acbd2d8787741bbbbf/src/Compilers/VisualBasic/Portable/Syntax/InternalSyntax/SyntaxToken.vb#L534-L537

**Pull request merged**